README.md

```diff
@@ -7,29 +7,310 @@
+```dart
+  Future<int> requestMtu(int mtu, {String transactionId}) {
+    return _manager.requestMtu(this, mtu, transactionId);
+  }
+```
+Request peripheral to set a different MTU. Operation is cancellable, meaning the operation's result in Dart can be discarded.
+Return the MTU set by peripheral after request.
+#### Characteristic convenience methods
+Following operations will be discussed in details **Characteristic** section.
+`CharacteristicWithValue` object is a `Characteristic` with additional `Uint8List value` property.
+If there are multiple service/characteristic UUID matches, runs the operation on the first discovered.
+```dart
+Future<CharacteristicWithValue> readCharacteristic(
+    String serviceUUID,
+    String characteristicUUID, {
+    String transactionId,
+  });
+```
+```dart
+  Future<Characteristic> writeCharacteristic(
+    String serviceUUID,
+    String characteristicUUID,
+    Uint8List bytes,
+    bool withResponse, {
+    String transactionId,
+  });
+```
+```dart
+  Stream<CharacteristicWithValue> monitorCharacteristic(
+    String serviceUUID,
+    String characteristicUUID, {
+    String transactionId,
+  });
+```
+### Service
+Object representing a unique service associated with unique peripheral, ensured by the internal mechanisms of MultiPlatformBleAdapter.
+For ease of use exposes property `peripheral`.
+#### Obtaining characteristics
+```dart
+  Future<List<Characteristic>> characteristics();
+```
+Returns a list of characteristics this service contains.
+#### Characteristic convenience methods
+Following operations will be discussed in details **Characteristic** section.
+`CharacteristicWithValue` object is a `Characteristic` with additional `Uint8List value` property.
 
+```dart
+  Future<Characteristic> writeCharacteristic(
+    String characteristicUUID,
+    Uint8List bytes,
+    bool withResponse, {
+    String transactionId,
+  });
+```
+```dart
+  Future<CharacteristicWithValue> readCharacteristic(String characteristicUUID,
+          {String transactionId});
+```
+```dart
+  Stream<CharacteristicWithValue> monitorCharacteristic(
+    String characteristicUUID, {
+    String transactionId,
+  });
+```
+
+### Characteristic
+Object representing unique characteristic inside a unique service associated with unique peripheral, all ensured by the internal mechanisms of MultiPlatformBleAdapter.
+For ease of use exposes property `service`.
+Contains following boolean properties:
+* `isReadable`
+* `isWritableWithResponse`
+* `isWritableWithoutResponse`
+* `isNotifiable`
+* `isIndicatable`
+
+If no `transactionId` has been specified, then it defaults to `null`. 
+**IMPORTANT** If previous operation with specified `transactionId` has not yet finished, it will be cancelled immediately.
+
+```dart
+  Future<Uint8List> read({String transactionId});
+```
+Reads the value of the characteristic. Operation is cancellable, meaning the operation's result in Dart can be discarded.
```

Comment:
There is no need to describe here how the cancellation mechanism works as it was already described above in the *Cancelling asynchronous operations* section